File: components_page/components/fade/transition.py
import dash_bootstrap_components as dbc
from dash import Input, Output, State, html

fade = html.Div(
    [
        dbc.Button(
            "Toggle fade",
            id="fade-transition-button",
            className="mb-3",
            n_clicks=0,
        ),
        dbc.Fade(
            dbc.Card(
                dbc.CardBody(
                    html.P(
                        "This content fades in and out", className="card-text"
                    )
                )
            ),
            id="fade-transition",
            is_in=True,
            style={"transition": "opacity 2000ms ease"},
            timeout=2000,
        ),
    ]
)


@app.callback(
    Output("fade-transition", "is_in"),
    [Input("fade-transition-button", "n_clicks")],
    [State("fade-transition", "is_in")],
)
def toggle_fade(n, is_in):
    if not n:
        # Button has never been clicked
        return True
    return not is_in

File: components_page/components/fade/transition.R
library(dashBootstrapComponents)

fade <- div(
  list(
    dbcButton("Toggle fade",
      id = "fade-transition-button", n_clicks = 0,
      className = "mb-3"
    ),
    dbcFade(
      dbcCard(
        dbcCardBody(
          p(
            "This content fades in and out",
            className = "card-text"
          )
        )
      ),
      id = "fade-transition",
      is_in = TRUE,
      style = list(transition = "opacity 2000ms ease"),
      timeout = 2000
    )
  )
)


app$callback(
  output("fade-transition", "is_in"),
  list(
    input("fade-transition-button", "n_clicks"),
    state("fade-transition", "is_in")
  ),
  function(n, is_in) {
    if (n > 0) {
      return(!is_in)
    }
    return(TRUE)
  }
)
